static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -238,6 +246,23 @@ function SelectProjectStep({
   );
 }
 
+function usePublicDSN({
+  organization,
+  project,
+}: {
+  organization: Organization | null;
+  project: Project | null;
+}) {
+  const response = useProjectKeys({organization, project});
+  if (response.type !== 'resolved') {
+    return response;
+  }
+  return {
+    ...response,
+    data: response.data[0]?.dsn.public ?? null,
+  };
```

Comment:
Returning a resolved response with a null `dsn` feels counter-intuitive. Should this explicitly return an error state instead?

Reply:
the only reason why i'm returning `null` here is to revert to the previous behaviour which is to render a simple ellipsis.

we can throw an error here, but i don't see a good reason to completely block onboarding in the odd event we couldn't load a `dsn`. is there any reason, besides some type of network anomaly there wouldn't be a `dsn`?